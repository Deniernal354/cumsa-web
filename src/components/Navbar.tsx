"use client";
import Link from "next/link";

import { usePathname } from "next/navigation";
import { useEffect, useRef, useState } from "react";
import { LINKS } from "@/lib/constants";
import { FaTree, FaChevronDown } from "react-icons/fa";
import { motion } from "framer-motion";

type NavItem = {
  label: string;
  href?: string;
  children?: { label: string; href: string }[];
};

const NAV: NavItem[] = [
  { label: "Home", href: "/" },
  { label: "Donate", href: "/donate" },
  {
    label: "About Us",
    children: [
      { label: "About Us", href: "/about" },
      { label: "Land Acknowledgment", href: "/land-acknowledgement" },
      { label: "Documents", href: "/documents" },
    ],
  },
  {
    label: "Services",
    children: [
      { label: "Prayer Services", href: "/services/prayer" },
      { label: "Chaplaincy Services", href: "/services/chaplaincy" },
      { label: "Roommate Services", href: "/services/roommate" },
      { label: "Equity Services", href: "/services/equity" },
      { label: "Your Masjid", href: "/services/your-masajid" },
      { label: "New Muslims", href: "/services/new-muslims" },
    ],
  },
  { label: "Join Us", href: "/join-us" },
  { label: "Contact Us", href: "/contact-us" },
];

export default function Navbar() {
  const pathname = usePathname();
  const [mobileOpen, setMobileOpen] = useState(false);
  const [openDropdown, setOpenDropdown] = useState<string | null>(null);
  const navRef = useRef<HTMLDivElement>(null);
  const [hoveredIdx, setHoveredIdx] = useState<number | null>(null);
  const navRefs = useRef<(HTMLDivElement | null)[]>([]);
  const [highlightProps, setHighlightProps] = useState({ left: 0, width: 0 });

  useEffect(() => {
    if (hoveredIdx !== null && navRefs.current[hoveredIdx]) {
      const el = navRefs.current[hoveredIdx]!;
      const rect = el.getBoundingClientRect();
      const parentRect = el.parentElement!.getBoundingClientRect();
      setHighlightProps({ left: rect.left - parentRect.left, width: rect.width });
    }
  }, [hoveredIdx]);

  useEffect(() => {
    setMobileOpen(false);
    setOpenDropdown(null);
  }, [pathname]);
  useEffect(() => {
    const onClick = (e: MouseEvent) => {
      if (!navRef.current) return;
      if (!navRef.current.contains(e.target as Node)) {
        setOpenDropdown(null);
      }
    };
    window.addEventListener("click", onClick);
    return () => window.removeEventListener("click", onClick);
  }, []);

  return (
    <div ref={navRef} className="sticky top-0 z-50 shadow">
      {}
      <div className="bg-white text-[var(--navy)]">
        <div className="container-base h-16 flex items-center justify-between">
          <Link
            href="/"
            className="flex items-center select-none"
            style={{ WebkitTapHighlightColor: "transparent" }}
          >
            <motion.img
              src="/MSA.png"
              alt="Carleton University MSA logo"
              width={220}
              height={56}
              className="object-contain h-14 w-auto drop-shadow-md"
              whileHover={{ scaleX: 1.12, scaleY: 0.92 }}
              whileTap={{ scale: 1.18 }}
              transition={{ type: "spring", stiffness: 350, damping: 12, duration: 0.3 }}
              draggable={false}
            />
          </Link>

          <button
            aria-label="Toggle menu"
            type="button"
            className="md:hidden p-2 rounded hover:bg-white/10 transition-base"
            onClick={() => setMobileOpen((v) => !v)}
          >
            <span className="block w-6 h-0.5 bg-[var(--navy)] mb-1" />
            <span className="block w-6 h-0.5 bg-[var(--navy)] mb-1" />
            <span className="block w-6 h-0.5 bg-[var(--navy)]" />
          </button>

          <nav className="hidden md:flex items-center gap-6 relative">
            <motion.div
              className="absolute top-0 h-full bg-gradient-to-br from-blue-200/60 to-blue-400/30 rounded-lg pointer-events-none z-0"
              animate={
                hoveredIdx !== null
                  ? { left: highlightProps.left, width: highlightProps.width, opacity: 1 }
                  : { opacity: 0 }
              }
              transition={{ type: "spring", stiffness: 400, damping: 30, duration: 0.25 }}
              style={{ left: highlightProps.left, width: highlightProps.width }}
            />
            {NAV.map((item, i) => (
              <div
                key={item.label}
                ref={(el) => {
                  navRefs.current[i] = el;
                }}
                className="relative z-10"
                onMouseEnter={() => setHoveredIdx(i)}
                onMouseLeave={() => setHoveredIdx(null)}
              >
                {"children" in item ? (
                  <>
                    <button
                      className={`flex items-center gap-1 px-2 py-1 rounded transition-colors duration-200 ${hoveredIdx === i ? "bg-blue-100/60 text-[var(--blue)] shadow" : "hover:bg-blue-50/60 hover:text-[var(--blue)]"}`}
                      type="button"
                      onClick={(e) => {
                        e.stopPropagation();
                        setOpenDropdown((d) => (d === item.label ? null : item.label));
                      }}
                      aria-expanded={openDropdown === item.label}
                      aria-haspopup="menu"
                    >
                      {item.label}
                      <FaChevronDown
                        className={`w-4 h-4 transition-transform ${openDropdown === item.label ? "rotate-180" : ""}`}
                        aria-hidden
                      />
                    </button>
                    {openDropdown === item.label && (
                      <div className="absolute left-0 mt-2 w-56 rounded-md bg-white text-[var(--navy)] shadow ring-1 ring-black/5 py-2">
                        {(item.children ?? []).map((c: { label: string; href: string }) =>
                          c.href?.startsWith("http") ? (
                            <a
                              key={c.href}
                              href={c.href}
                              target="_blank"
                              rel="noopener noreferrer"
                              className="block px-4 py-2 hover:bg-[var(--blue)] hover:text-white transition-base"
                            >
                              {c.label}
                            </a>
                          ) : (
                            <Link
                              key={c.href}
                              href={c.href}
                              className="block px-4 py-2 hover:bg-[var(--blue)] hover:text-white transition-base"
                            >
                              {c.label}
                            </Link>
                          ),
                        )}
                      </div>
                    )}
                  </>
                ) : (
                  "href" in item &&
                  item.href && (
                    <Link
                      href={item.href}
                      className={`px-2 py-1 rounded transition-colors duration-200 ${hoveredIdx === i ? "bg-blue-100/60 text-[var(--blue)] shadow" : "hover:bg-blue-50/60 hover:text-[var(--blue)]"} ${pathname === item.href ? "text-[var(--blue)]" : ""}`}
                    >
                      {item.label}
                    </Link>
                  )
                )}
              </div>
            ))}
            <a
              href={LINKS.linktree}
              target="_blank"
              rel="noopener noreferrer"
              aria-label="Linktree"
              className="ml-2 p-2 rounded-full hover:bg-green-100 transition-base text-green-700 hover:text-green-800 z-10"
            >
              <FaTree className="w-5 h-5" />
            </a>
          </nav>
        </div>
      </div>

      {mobileOpen && (
        <div className="md:hidden border-t border-black/10 bg-white">
          <div className="container-base py-2 space-y-1">
            {NAV.map((item) => (
              <div key={item.label} className="">
                {"children" in item ? (
                  <details className="group">
                    <summary className="list-none flex items-center justify-between py-2 cursor-pointer">
                      <span>{item.label}</span>
                      <FaChevronDown
                        className="w-4 h-4 transition group-open:rotate-180"
                        aria-hidden
                      />
                    </summary>
                    <div className="pl-4 pb-2 space-y-1">
                      {(item.children ?? []).map((c: { label: string; href: string }) =>
                        c.href?.startsWith("http") ? (
                          <a
                            key={c.href}
                            href={c.href}
                            target="_blank"
                            rel="noopener noreferrer"
                            className="block py-1 hover:text-[var(--blue)]"
                          >
                            {c.label}
                          </a>
                        ) : (
                          <Link
                            key={c.href}
                            href={c.href}
                            className="block py-1 hover:text-[var(--blue)]"
                          >
                            {c.label}
                          </Link>
                        ),
                      )}
                    </div>
                  </details>
                ) : (
                  "href" in item &&
                  item.href && (
                    <Link href={item.href} className="block py-2 hover:text-[var(--blue)]">
                      {item.label}
                    </Link>
                  )
                )}
              </div>
            ))}
          </div>
        </div>
      )}
    </div>
  );
}
