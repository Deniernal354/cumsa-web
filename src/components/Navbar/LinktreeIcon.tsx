import { TbBrandLinktree } from "react-icons/tb";
import { motion } from "framer-motion";
import { LINKS } from "@/lib/constants";

const circleVariants = {
  idle: { scale: 0.8, opacity: 0, backgroundColor: "rgba(16,185,129,0)" }, // transparent
  hover: { scale: 1, opacity: 1, backgroundColor: "rgba(16,185,129,0.98)" }, // teal/green
};

const iconVariants = {
  idle: { color: "#059669" }, // green-600
  hover: { color: "#ffffff" },
};

export default function LinktreeIcon() {
  return (
    <motion.a
      href={LINKS.linktree}
      target="_blank"
      rel="noopener noreferrer"
      aria-label="Linktree"
      className="relative inline-flex items-center justify-center w-9 h-9 rounded-full"
      initial="idle"
      whileHover="hover"
      whileTap="idle"
    >
      {/* circular green background */}
      <motion.span
        className="absolute z-0 rounded-full"
        style={{ width: 36, height: 36, borderRadius: 9999 }}
        variants={circleVariants}
        transition={{ duration: 0.14 }}
      />

      <motion.span
        className="relative z-10 flex items-center justify-center"
        variants={iconVariants}
        transition={{ duration: 0.12 }}
      >
        <TbBrandLinktree className="w-5 h-5" />
      </motion.span>
    </motion.a>
  );
}
