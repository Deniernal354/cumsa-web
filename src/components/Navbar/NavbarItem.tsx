import Link from "next/link";
import { NavItem } from "./navData";

interface NavbarItemProps {
  item: NavItem;
  active: boolean;
}

export default function NavbarItem({ item, active }: NavbarItemProps) {
  if (!item.href) return null;

  return (
    <div className="relative inline-block">
      <Link
        href={item.href}
        className={`px-3 py-2 rounded-lg transition-colors duration-150 relative z-10 font-medium ${active ? "text-[var(--blue)]" : "text-slate-700 hover:text-[var(--blue)]"}`}
      >
        {item.label}
      </Link>

      {active && (
        <span className="absolute -bottom-2 left-1/2 transform -translate-x-1/2 w-8 h-1 rounded-full bg-[var(--blue)] z-20" />
      )}
    </div>
  );
}

interface MobileNavItemProps {
  item: NavItem;
}

export function MobileNavItem({ item }: MobileNavItemProps) {
  if (item.href) {
    return (
      <Link
        href={item.href}
        className="block py-2 px-4 hover:text-[var(--blue)] hover:bg-[color:rgba(90,206,216,0.12)] rounded-lg transition-all duration-200"
      >
        {item.label}
      </Link>
    );
  }
  return null;
}
