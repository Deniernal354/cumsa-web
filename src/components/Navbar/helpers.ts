import { useEffect, useRef, useCallback, useState } from "react";

export const NAV_Z_INDEX = 9999;

export const animationVariants = {
  dropdown: {
    initial: { opacity: 0, y: -10, scale: 0.95 },
    animate: { opacity: 1, y: 0, scale: 1 },
    exit: { opacity: 0, y: -10, scale: 0.95 },
    transition: { type: "spring" as const, stiffness: 500, damping: 30, duration: 0.2 },
  },
  logo: {
    whileHover: { scaleX: 1.12, scaleY: 0.92 },
    whileTap: { scale: 1.18 },
    transition: { type: "spring" as const, stiffness: 350, damping: 12, duration: 0.3 },
  },
};

export function useClickOutside(ref: React.RefObject<HTMLDivElement | null>, callback: () => void) {
  useEffect(() => {
    const handleClick = (e: MouseEvent) => {
      if (!ref.current?.contains(e.target as Node)) {
        callback();
      }
    };

    window.addEventListener("click", handleClick);
    return () => window.removeEventListener("click", handleClick);
  }, [ref, callback]);
}

export function useHighlight(
  hoveredIdx: number | null,
  navRefs: React.MutableRefObject<(HTMLDivElement | null)[]>,
) {
  const getHighlightProps = useCallback(() => {
    if (hoveredIdx !== null && navRefs.current[hoveredIdx]) {
      const el = navRefs.current[hoveredIdx]!;
      const rect = el.getBoundingClientRect();
      const parentRect = el.parentElement!.getBoundingClientRect();
      return {
        left: rect.left - parentRect.left,
        width: rect.width,
      };
    }
    return { left: 0, width: 0 };
  }, [hoveredIdx, navRefs]);

  return getHighlightProps();
}

export function useNavbarState() {
  const navRef = useRef<HTMLDivElement>(null);
  const navRefs = useRef<(HTMLDivElement | null)[]>([]);

  return {
    navRef,
    navRefs,
  };
}

export function useLiquidCursor(navRef: React.RefObject<HTMLDivElement | null>) {
  const [cursorPosition, setCursorPosition] = useState({ x: 0, y: 0 });
  const [isHovering, setIsHovering] = useState(false);

  useEffect(() => {
    const handleMouseMove = (e: MouseEvent) => {
      if (!navRef.current) return;

      const rect = navRef.current.getBoundingClientRect();
      const x = e.clientX - rect.left;
      const y = e.clientY - rect.top;

      setCursorPosition({ x, y });
    };

    const handleMouseEnter = () => setIsHovering(true);
    const handleMouseLeave = () => setIsHovering(false);

    const navElement = navRef.current;
    if (navElement) {
      navElement.addEventListener("mousemove", handleMouseMove);
      navElement.addEventListener("mouseenter", handleMouseEnter);
      navElement.addEventListener("mouseleave", handleMouseLeave);
    }

    return () => {
      if (navElement) {
        navElement.removeEventListener("mousemove", handleMouseMove);
        navElement.removeEventListener("mouseenter", handleMouseEnter);
        navElement.removeEventListener("mouseleave", handleMouseLeave);
      }
    };
  }, [navRef]);

  return { cursorPosition, isHovering };
}
