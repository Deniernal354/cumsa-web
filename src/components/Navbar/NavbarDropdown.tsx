import Link from "next/link";
import { motion } from "framer-motion";
import { FaChevronDown } from "react-icons/fa";
import { NavItem } from "./navData";
import { NAV_Z_INDEX } from "./helpers";

interface NavbarDropdownProps {
  item: NavItem;
  isOpen: boolean;
  active?: boolean;
  onToggle: () => void;
}

export default function NavbarDropdown({ item, isOpen, active, onToggle }: NavbarDropdownProps) {
  return (
    <div className="relative">
      {/* show a small blue block when active */}
      {active && (
        <span className="absolute -bottom-2 left-1/2 transform -translate-x-1/2 w-8 h-1 rounded-full bg-blue-600 z-0" />
      )}

      <button
        className={`
          flex items-center gap-2 px-3 py-2 rounded-lg transition-all duration-200 relative z-20 font-medium
          ${isOpen ? "text-blue-600" : "text-slate-700 hover:text-blue-600"}
        `}
        type="button"
        onClick={(e) => {
          e.stopPropagation();
          onToggle();
        }}
        onMouseEnter={() => {}}
        onMouseLeave={() => {}}
        aria-expanded={isOpen}
        aria-haspopup="menu"
      >
        <span>{item.label}</span>
        <motion.div animate={{ rotate: isOpen ? 180 : 0 }} transition={{ duration: 0.2 }}>
          <FaChevronDown className="w-3 h-3" aria-hidden />
        </motion.div>
      </button>

      {isOpen && (
        <motion.div
          className="absolute left-0 top-full mt-2 w-56 rounded-lg bg-white shadow-lg ring-1 ring-black/5 py-1 border border-gray-100"
          initial={{ opacity: 0, y: -8, scale: 0.95 }}
          animate={{ opacity: 1, y: 0, scale: 1 }}
          exit={{ opacity: 0, y: -8, scale: 0.95 }}
          transition={{ duration: 0.15 }}
          style={{ zIndex: NAV_Z_INDEX + 1 }}
        >
          {(item.children ?? []).map((child) => (
            <DropdownItem key={child.href} child={child} />
          ))}
        </motion.div>
      )}
    </div>
  );
}

interface DropdownItemProps {
  child: { label: string; href: string };
}

function DropdownItem({ child }: DropdownItemProps) {
  const itemClass =
    "block px-4 py-2 hover:bg-blue-50 hover:text-blue-600 transition-colors duration-150 rounded-md mx-2 my-1";

  if (child.href.startsWith("http")) {
    return (
      <a href={child.href} target="_blank" rel="noopener noreferrer" className={itemClass}>
        {child.label}
      </a>
    );
  }

  return (
    <Link href={child.href}>
      <div className={itemClass}>{child.label}</div>
    </Link>
  );
}

interface MobileDropdownProps {
  item: NavItem;
}

export function MobileDropdown({ item }: MobileDropdownProps) {
  return (
    <details className="group">
      <summary className="list-none flex items-center justify-between py-2 cursor-pointer">
        <span>{item.label}</span>
        <FaChevronDown className="w-4 h-4 transition group-open:rotate-180" aria-hidden />
      </summary>
      <div className="pl-4 pb-2 space-y-1">
        {(item.children ?? []).map((child) => (
          <MobileDropdownItem key={child.href} child={child} />
        ))}
      </div>
    </details>
  );
}

function MobileDropdownItem({ child }: { child: { label: string; href: string } }) {
  if (child.href.startsWith("http")) {
    return (
      <a
        href={child.href}
        target="_blank"
        rel="noopener noreferrer"
        className="block py-1 hover:text-[var(--blue)]"
      >
        {child.label}
      </a>
    );
  }

  return (
    <Link href={child.href} className="block py-1 hover:text-[var(--blue)]">
      {child.label}
    </Link>
  );
}
