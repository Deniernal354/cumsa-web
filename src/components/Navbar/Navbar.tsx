"use client";
import Link from "next/link";
import { usePathname } from "next/navigation";
import { useEffect, useState, useCallback } from "react";
import { motion } from "framer-motion";
import { NAV } from "./navData";
import NavbarItem, { MobileNavItem } from "./NavbarItem";
import NavbarDropdown, { MobileDropdown } from "./NavbarDropdown";
import LinktreeIcon from "./LinktreeIcon";
import { useClickOutside, useNavbarState, NAV_Z_INDEX } from "./helpers";

export default function Navbar() {
  const pathname = usePathname();
  const [mobileOpen, setMobileOpen] = useState(false);
  const [openDropdown, setOpenDropdown] = useState<string | null>(null);

  const { navRef, navRefs } = useNavbarState();

  const activeIndex = NAV.findIndex((item) => {
    if (item.href && pathname === item.href) return true;
    if (item.children) {
      return item.children.some((c) => pathname === c.href || pathname.startsWith(c.href));
    }
    return false;
  });

  const closeAll = useCallback(() => {
    setOpenDropdown(null);
  }, []);

  useClickOutside(navRef, closeAll);

  useEffect(() => {
    setMobileOpen(false);
    closeAll();
  }, [pathname, closeAll]);

  return (
    <div
      ref={navRef}
      style={{ zIndex: NAV_Z_INDEX }}
      className="sticky top-0 shadow-sm bg-white/95 backdrop-blur-md border-b border-gray-200"
    >
      <div className="text-slate-800 relative z-10">
        <div className="container-base h-16 flex items-center justify-between">
          <Link
            href="/"
            className="flex items-center select-none"
            style={{ WebkitTapHighlightColor: "transparent" }}
          >
            <motion.img
              src="/MSA.png"
              alt="Carleton University MSA logo"
              width={220}
              height={56}
              className="object-contain h-14 w-auto drop-shadow-sm"
              whileHover={{ scale: 1.08 }}
              whileTap={{ scale: 0.92 }}
              transition={{ type: "spring", stiffness: 340, damping: 18, mass: 0.6 }}
              draggable={false}
            />
          </Link>

          <div className="md:hidden flex items-center gap-3">
            <LinktreeIcon />
            <button
              aria-label="Toggle menu"
              type="button"
              className="p-2 rounded-lg hover:bg-gray-100 transition-colors duration-200"
              onClick={() => setMobileOpen((v) => !v)}
            >
              <span className="block w-6 h-0.5 bg-slate-600 mb-1" />
              <span className="block w-6 h-0.5 bg-slate-600 mb-1" />
              <span className="block w-6 h-0.5 bg-slate-600" />
            </button>
          </div>

          <nav className="hidden md:flex items-center gap-1 relative">
            {/* Simple highlight background */}
            <motion.div
              className="absolute top-1 bottom-1 rounded-lg pointer-events-none z-0 bg-blue-100/60"
              animate={{ opacity: 0 }}
              transition={{ duration: 0.2 }}
            />

            {NAV.map((item, i) => {
              const isOpen = openDropdown === item.label;

              return (
                <div
                  key={item.label}
                  ref={(el) => {
                    navRefs.current[i] = el;
                  }}
                  className="relative z-10"
                  onMouseEnter={() => {}}
                  onMouseLeave={() => {}}
                >
                  {item.children ? (
                    <NavbarDropdown
                      item={item}
                      isOpen={isOpen}
                      active={i === activeIndex}
                      onToggle={() => setOpenDropdown(isOpen ? null : item.label)}
                    />
                  ) : (
                    <NavbarItem item={item} active={i === activeIndex} />
                  )}
                </div>
              );
            })}

            <div
              ref={(el) => {
                navRefs.current[NAV.length] = el;
              }}
              onMouseEnter={() => {}}
              onMouseLeave={() => {}}
              className="relative z-10 ml-2"
            >
              <LinktreeIcon />
            </div>
          </nav>
        </div>
      </div>

      {mobileOpen && (
        <motion.div
          className="md:hidden border-t border-gray-200 bg-white"
          initial={{ opacity: 0, height: 0 }}
          animate={{ opacity: 1, height: "auto" }}
          exit={{ opacity: 0, height: 0 }}
          transition={{ duration: 0.2 }}
        >
          <div className="container-base py-4 space-y-1">
            {NAV.map((item) => (
              <div key={item.label}>
                {item.children ? <MobileDropdown item={item} /> : <MobileNavItem item={item} />}
              </div>
            ))}
          </div>
        </motion.div>
      )}
    </div>
  );
}
