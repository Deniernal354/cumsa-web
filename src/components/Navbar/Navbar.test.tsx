import { render } from "@testing-library/react";
import Navbar from "./Navbar";

test("Navbar snapshot", () => {
  const { asFragment } = render(<Navbar />);
  expect(asFragment()).toMatchSnapshot();
});
