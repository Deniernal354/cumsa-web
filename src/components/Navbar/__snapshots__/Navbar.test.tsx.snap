// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<DocumentFragment>
  <nav
    class="w-full bg-white/90 backdrop-blur border-b border-black/10 sticky top-0 z-50"
  >
    <div
      class="container-base flex items-center justify-between h-16"
    >
      <a
        class="flex items-center gap-2"
        href="/"
      >
        <img
          alt="CUMSA logo"
          class="object-contain rounded-sm h-8 w-auto"
          height="44"
          src="/CU-MSA-LOGO.png"
          style="width: 44px; height: 44px;"
          width="44"
        />
      </a>
      <ul
        class="flex gap-4 items-center"
      >
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/services/prayer"
          >
            Prayers
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/services/chaplaincy"
          >
            Chaplaincy
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/services/roommate"
          >
            Roommate
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/services/new-muslims"
          >
            New Muslims
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/documents"
          >
            Documents
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/donate"
          >
            Donate
          </a>
        </li>
        <li>
          <a
            class="text-sm font-medium text-[var(--navy)] hover:text-[var(--blue)] transition-colors"
            href="/contact-us"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
